### ISSUE TITLE
IntelliSense squiggles randomly stop updating after editing

### ISSUE THREAD

**Issue by sean-mcmanus** [microsoft/vscode-cpptools#7118]
I've been hitting this a lot recently.

Squiggle updates are being blocked because it thinks it's in the middle of an existing update when it's not.

Not sure of an easy repro yet -- is anyone able to figure out any repro steps?

I've also seen that outline view (tag parsing) not updating as it should randomly, might be unrelated.

**Comment by v-ericawu** [comment]
Hi, @sean-mcmanus I am not sure your issue is same as mine, please check my steps. If they are different with yours, please share the demo project and repro steps to us.
My steps:
1. Install Windows OS with 19042.804  build .
2. Install VS Code 1.54.1 version.
3. Install C/C++ Extension 1.1.3 version.
4. Open project Folder. 
5. Type the error message in main.cpp and appears Squiggle. 
6. Updates C/C++ Extension from 1.1.3 to 1.2.2 version.

**Comment by v-ericawu** [comment]
@sean-mcmanus Sorry, we didn't repro this issue and we just listed our tried scenarios as above. We will update the repro steps once we meet this issue in future.

**Comment by patrikanderssononsala** [comment]
We are seeing this issue quite frequently at our company. We have a fairly large project and I have not been able to narrow it down to a smaller project. I cannot share the project either. :-(

I have however a certain procedure now in our code that always triggers bug. I'm a total noob on how to debug these things in vs code. I tried to enable C_Cpp.loggingLevel but it really does not provide any help - would have liked to see some kind of error message or something there.

When I do "reload window" the error goes away.

**Comment by sean-mcmanus** [comment]
@patrikanderssononsala If a particular C/C++ code construct is triggering the symptom for you, then that is a different root cause from what I hit. In your case, it sounds like the particular code is causing a failure in our IntelliSense parser, possibly a crash if the C/C++ log has a "crash" message. We might need some sort of code sample for the repro.

In the case I was hitting, it appears to be triggered via some random timing/editing and isn't reproduceable later with the same code.

**Comment by patrikanderssononsala** [comment]
@sean-mcmanus OK, sorry for being quite short in my message, will try to explain better:
It is **not** a particular code c++ construct, but it was rather that everytime I created erroneous code, the squiggles that showed up were there to stay. So procedure was like this:
1) Start vs code - no squiggles. 
2) Modify code in an erroneous way (spelling mistake, etc). Squiggles show up.
3) Undo changes back to OK code. Squiggles there still, including error message about the error that was there before.

I can add the following also:
4) "Reload window" or restart vs code (or computer...) - squiggles goes away (I have been monitoring the output from the `C_Cpp.loggingLevel = Debug` and I know for sure that the indexing did reach a steady state, so my file was finished)
5-6-7) Same as 1-2-3 above, i.e. workspace goes immediately back into erroneous state once I start editing.

New info:
This time I tried to "Reset intellisense database" and that actually cured the workspace. I have done that before, but the error has come back then. So I am quite sure it will come back again. Would there be any logging that you would like me to have enabled if that happens? I will leave the `C_Cpp.loggingLevel` on (it is comforting to see how the indexer is working hard to index our quite large repo). Anything else?

In your case, how do you get out of the situation?

**Comment by sean-mcmanus** [comment]
After step 3, you may need to move your cursor to a new line in order for the squiggle to update (which is "by design") -- does the squiggle still stays after moving the cursor to a new line?

Hmm...Reset IntelliSense Database will clear squiggles, but I don't know how it could cause or fix the stuck squiggles.

In my case (where the "updating" state is incorrect stuck), doing a reload window will fix it.

**Comment by patrikanderssononsala** [comment]
Yes, the squiggles stayed in the erroneous mode until "reload window", including moving mouse. I actually have worked quite a while with vs code in this erroneous mode so I am quite sure now it does not get out of it.

It feels like the data base was in some kind of erroneous state. 

Somewhat similar errors - but at the same time maybe a bit different? Mine was 100% reproducible until I reset the intellisense database. Yours is a bit random. Hmmm.
